apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/ContextManager.java

```diff
@@ -165,12 +166,20 @@ public static AbstractSpan activeSpan() {
         return get().activeSpan();
     }
 
+    /**
+    * Plese use `ContextManager.stopSpan(span)` if could, it's can "fail-fast"
```

Comment:
Why call it `fail-fast`? From Wikipedia, this is the definition of it. At here, you are just avoiding to get threadlocal twice. I think you should re-write the comment.

> In systems design, a fail-fast system is one which immediately reports at its interface any condition that is likely to indicate a failure. Fail-fast systems are usually designed to stop normal operation rather than attempt to continue a possibly flawed process.

Reply:
This comment is target another design issue:
use `ContextManager.stopSpan(span)` better than 'ContextManager.stopSpan()`
it's found program error 
ie:
```
        final AbstractSpan span10 = ContextManager.createLocalSpan("op10");
        //work10
        final AbstractSpan span11 = ContextManager.createLocalSpan("op11");
      //work11
        ContextManager.stopSpan(span10); 
      //...
        ContextManager.stopSpan(span11);
```
if  'ContextManager.stopSpan(span10);' is plugin bug, should not call stop in this point
then will throw: ` java.lang.IllegalStateException: Stopping the unexpected`
I call it's a Fail-fast design

If just call `ContextManager.stopSpan()` will hide the bug in the plugin

不知这样英文表达是否清楚，我中文再说了一下吧：
 `ContextManager.stopSpan()` 是容易隐藏编程错误 
而明确调用 'ContextManager.stopSpan(span10);' 如顺序不对就会直接报错-- Fail-fast

Reply:
This thing, you are describing is already in codes, so it is not related to this PR. 
I could get your point, but you should know, many plugins can't choose this way, even they want to. 

Fail-fast is a concept widely definition, you should not define it, especially for an English term.

If you want to add some comments or recommendation of using this, raise other PR with more accurate words.

Reply:
Also, please notice, unless all plugins use `stopSpan(span)` API, the error is still uncertain. You can't tell which plugin triggers it.

Reply:
The reality is, they wouldn't be going to change because it is impossible.